### PR TITLE
Plane: fixed use of THR_MIN when disarmed

### PR DESCRIPTION
**Pull request by tridge**
Don't use THR_MIN when disarmed unless on an IC system with throttle while disarmed set

### CODE REVIEW

File: libraries/AP_ICEngine/AP_ICEngine.h [hunk 1]
```diff
@@ -55,6 +55,11 @@ class AP_ICEngine {
     // update min throttle for idle governor
     void update_idle_governor(int8_t &min_throttle);
 
+    // do we have throttle while disarmed enabled?
+    bool allow_throttle_while_disarmed(void) const {
+        return option_set(Options::THROTTLE_WHILE_DISARMED);
```
Comment (IamPete1) [inline libraries/AP_ICEngine/AP_ICEngine.h:60]:
```suggestion
        return enable && option_set(Options::THROTTLE_WHILE_DISARMED);
```

File: ArduPlane/servos.cpp [hunk 2]
```diff
@@ -890,7 +890,11 @@ void Plane::set_servos(void)
     // slew rate limit throttle
     throttle_slew_limit(SRV_Channel::k_throttle);
 
+    int8_t min_throttle = 0;
 #if AP_ICENGINE_ENABLED
+    if (g2.ice_control.allow_throttle_while_disarmed()) {
+        min_throttle = MAX(aparm.throttle_min.get(), 0);
+    }
```
Comment (IamPete1) [inline ArduPlane/servos.cpp:897]: I think this could still be down in the switch statement?

**Review by IamPete1 — COMMENTED** [review]
I'm not sure why this case is missed by the `throttle_override` call? Maybe I have miss understood the issue.

File: libraries/AP_ICEngine/AP_ICEngine.h (hunk 1 above)
Comment (tridge) [reply libraries/AP_ICEngine/AP_ICEngine.h]: fixed, thanks

File: ArduPlane/servos.cpp (hunk 2 above)
Comment (tridge) [reply ArduPlane/servos.cpp:897]: could, but then we have 3 ugly ifdefs .... I think its nicer with 2

**Comment by tridge** [comment]
> I'm not sure why this case is missed by the `throttle_override` call? Maybe I have miss understood the issue.

the issue came up for someone with an electric aircraft, ICE disabled. Setting THR_MIN led to non-zero throttle when disarmed. So it is quite a nasty issue

**Comment by IamPete1** [comment]
Ok, I understand, the problem is without ICE, not with. 

[ARMING_REQUIRE](https://ardupilot.org/plane/docs/parameters.html#arming-require-require-arming-motors) specifically says: "1: THR_MIN PWM when disarmed" 

So the current behaviour is correct to the param description.

I think we should:
1. Remove the override in `sevos.cpp` all together changing ARMING_REQUIRE description to be 0 throttle when disarmed. 

2. Move the override of throttle min into the ICE lib within its existing `throttle_override` function. Possibly even with a new ICE param. Using the existing one for this special ICE use case is a little odd.

**Comment by tridge** [comment]
> 2\. Move the override of throttle min into the ICE lib within its existing `throttle_override` function.

no, we still need THR_MIN to be active when armed on plane without ICE, it would be too big a change to get rid of it completely for non-ICE planes

**Comment by IamPete1** [comment]
I agree we need the min when armed in all cases, just this special ICE disarmed case is a little odd when there is already a function for ICE to override throttle.

**Review by IamPete1 — CHANGES_REQUESTED** [review]
Needs update to `ARMING_REQUIRE` description to reflect this new behaviour. 

Moving this functionality down into ICE would also be nice.

**Comment by tridge** [comment]
> Needs update to `ARMING_REQUIRE` description to reflect this new behaviour.

done

> Moving this functionality down into ICE would also be nice.

that doesn't really make sense as THR_MIN is also use with non-ICE planes. We could give ICE access to THR_MIN and do it for the disarmed case, but I don't think it will really be any simpler

**Review by IamPete1 — APPROVED** [review]

**Pull request merged**